store-cli_test.go

```diff
@@ -1,744 +1,55 @@
-	}
-
-	testKey = "foo[[]]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo[1e]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo[01]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo."
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo.[]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "-foo"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo-[]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo.-bar"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-
-	testKey = "foo.bar-[]"
-	if r = validateMetaKey(testKey); r == true {
-		t.Fatalf("'%v' is should be rejected", testKey)
-	}
-}
-
-func TestIndexOfFirstRightBracket(t *testing.T) {
-	key := "foo[1]"
-	i := indexOfFirstRightBracket(key)
-	expected := 5
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
-	}
-
-	key = "foo[10]"
-	i = indexOfFirstRightBracket(key)
-	expected = 6
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
-	}
-}
-
-func TestMetaIndexFromKey(t *testing.T) {
-	key := "foo[1]"
-	i := metaIndexFromKey(key)
-	expected := 1
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
-	}
-
-	key = "foo[10]"
-	i = metaIndexFromKey(key)
-	expected = 10
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
-	}
-
-	key = "foo[10].bar[4].baz"
-	i = metaIndexFromKey(key)
-	expected = 10
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
-	}
-
-	key = "foo[]"
-	i = metaIndexFromKey(key)
-	expected = 0
-
-	if i != expected {
-		t.Fatalf("Expected '%d' but '%d'", expected, i)
+func TestMakeURL(t *testing.T) {
+	os.Setenv("SD_STORE_URL", "http://store.screwdriver.cd")
+	os.Setenv("SD_BUILD_ID", "10038")
+	os.Setenv("SD_EVENT_ID", "499")
+
+	storeType := "cache"
+	scope := "events"
+	key := "cache-1"
+	i, _ := makeURL(storeType, scope, key)
+	expected := "http://store.screwdriver.cd/v1/caches/events/499/cache-1"
+	if i.String() != expected {
+		t.Fatalf("Expected '%s' but '%s'", expected, i)
+	}
+
+	storeType = "artifacts"
+	scope = "builds"
+	key = "test"
+	i, _ = makeURL(storeType, scope, key)
+	expected = "http://store.screwdriver.cd/v1/builds/10038-test"
+	if i.String() != expected {
+		t.Fatalf("Expected '%s' but '%s'", expected, i)
+	}
+
+	storeType = "artifacts"
+	scope = "events"
+	key = "test"
+	i, _ = makeURL(storeType, scope, key)
+	expected = "http://store.screwdriver.cd/v1/builds/events/499/test"
```

Comment:
Interesting. think we need to add some extra validation. 
If folder is `builds`, then it shouldn't append the `scope`. 
This combination shouldn't be allowed. This case should be `builds/{buildID}-ARTIFACTS`